### PR TITLE
Migrate deployment from (docker-compose, docker-swarm) to (Tilt, Kubernetes)

### PR DESCRIPTION
**Pull request by jazeved0**
## Description

This PR contains all of the changes needed to run the Architus app on Kubernetes in production. Additionally, includes a configuration for Tilt to run the app locally also in Kubernetes with features like hot-reloading.

## Remaining tasks

- Test the deployment to make sure it works (i.e. does this PR break any recent additions to develop such as lavalink or sandbox changes?)
- Make sure there are no unexpected changes in the PR

### CODE REVIEW

File: Tiltfile [hunk 1]
```diff
@@ -0,0 +1,81 @@
+load('ext://restart_process', 'docker_build_with_restart')
+
+optional_features = {
+    'feature-gate': ['feature-gate'],
+    'gateway': ['gateway'],
+    'api': ['api'],
+}
```
Comment (jazeved0) [inline Tiltfile]: TODO: backport changes in feature/logs-service to this PR, this tiltfile was better there

File: Tiltfile [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+k8s_yaml('db/kube/dev/db.yaml')
+k8s_yaml('sandbox/kube/dev/sandbox.yaml')
+k8s_yaml('dbmanager/kube/dev/dbmanager.yaml')
+
+k8s_resource('postgres', port_forwards=5432)
+k8s_resource('dbmanager')
+k8s_resource('shard')
+k8s_resource('sandbox', port_forwards=1337)
+k8s_resource('manager')
+k8s_resource('rabbit')
```
Comment (jazeved0) [inline Tiltfile]: todo include lavalink, redis

File: feature-gate/src/bin/feature-gate.rs [hunk 3]
```diff
@@ -62,7 +57,7 @@ impl FeatureGate for Gate {
 
         match result {
             Ok(_) => {
-                info!(
+                log::info!(
```
Comment (jazeved0) [inline feature-gate/src/bin/feature-gate.rs]: todo remove these changes

File: feature-gate/tilt-build/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Create minimal deployment
+FROM debian:buster-slim
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update -q \
+    && apt-get install -y -q "libssl1.1" "tini" "libpq-dev" \
```
Comment (jazeved0) [inline feature-gate/tilt-build/Dockerfile]: todo remove tini from this

File: lavalink/kube/prod/lavalink.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+apiVersion: apps/v1
```
Comment (jazeved0) [inline lavalink/kube/prod/lavalink.yaml:1]: todo add lavalink dev deployment

File: manager/manager_server.py [hunk 6]
```diff
@@ -32,9 +32,9 @@ def __init__(self, total_shards):
 
     def health_check(self):
         while True:
-            time.sleep(5)
+            time.sleep(1)
```
Comment (jazeved0) [inline manager/manager_server.py]: todo remove these changes

File: redis/kube/prod/redis.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+apiVersion: apps/v1
```
Comment (jazeved0) [inline redis/kube/prod/redis.yaml:1]: todo add development redis deployment

File: sandbox/kube/dev/sandbox.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  template:
+    metadata:
+      labels:
+        app: sandbox
+    spec:
+      containers:
+        - name: sandbox
+          image: sandbox-image
+          envFrom:
+          - secretRef:
```
Comment (jazeved0) [inline sandbox/kube/dev/sandbox.yaml]: TODO add environment variables here (`HTTP_PROXY` and `HTTPS_PROXY`)

File: sandbox/kube/prod/sandbox.yaml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        app: sandbox
+    spec:
+      containers:
+        - name: sandbox
+          image: "{{prefix}}sandbox:{{tag}}"
+          resources:
+            limits:
+              memory: "1Gi"
+              cpu: "1"
+          envFrom:
```
Comment (jazeved0) [inline sandbox/kube/prod/sandbox.yaml:23]: TODO add environment variables here (`HTTP_PROXY` and `HTTPS_PROXY`)

File: sandbox/sandbox.yaml [hunk 10]
```diff
@@ -0,0 +1,34 @@
+apiVersion: apps/v1
```
Comment (jazeved0) [inline sandbox/sandbox.yaml]: what is this file doing?

File: secret.example.yaml [hunk 11]
```diff
@@ -0,0 +1,18 @@
+# Fill out and rename to secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: architus-secret
+stringData:
+  NUM_SHARDS: "1"
+  bot_token:
+  db_user: "autbot"
+  db_pass: "autism"
```
Comment (jazevedo620) [inline secret.example.yaml:10]: weirdchamp

File: .gitignore [hunk 12]
```diff
@@ -14,4 +14,7 @@ shard/res/meme.png
 tags
 .vs/*
 .vscode/*
+.idea/*
 www
+secret.yaml
+tilt_modules
```
Comment (jazeved0) [inline .gitignore]: todo: remove this, it should not be ignored.

Also, todo: add tilt_modules to linguist-vendored

Comment (jazeved0) [reply .gitignore]: done

File: feature-gate/src/bin/feature-gate.rs (hunk 3 above)
Comment (jazeved0) [reply feature-gate/src/bin/feature-gate.rs]: done

File: feature-gate/tilt-build/Dockerfile (hunk 4 above)
Comment (jazeved0) [reply feature-gate/tilt-build/Dockerfile]: done

File: manager/manager_server.py (hunk 6 above)
Comment (jazeved0) [reply manager/manager_server.py]: done

File: Tiltfile (hunk 1 above)
Comment (jazeved0) [reply Tiltfile]: done

File: sandbox/kube/dev/sandbox.yaml (hunk 8 above)
Comment (jazeved0) [reply sandbox/kube/dev/sandbox.yaml]: done

File: sandbox/kube/prod/sandbox.yaml (hunk 9 above)
Comment (jazeved0) [reply sandbox/kube/prod/sandbox.yaml:23]: done

File: sandbox/sandbox.yaml (hunk 10 above)
Comment (jazeved0) [reply sandbox/sandbox.yaml]: removed

File: Tiltfile (hunk 2 above)
Comment (jazeved0) [reply Tiltfile]: done

File: redis/kube/prod/redis.yaml (hunk 7 above)
Comment (jazeved0) [reply redis/kube/prod/redis.yaml:1]: done

File: lavalink/kube/prod/lavalink.yaml (hunk 5 above)
Comment (jazeved0) [reply lavalink/kube/prod/lavalink.yaml:1]: done

**Review by johnyburd — APPROVED** [review]
excellent work everyone :)

**Pull request merged**